.pre-commit-config.yaml

```diff
@@ -12,49 +12,24 @@ repos:
           - id: trailing-whitespace
             exclude: miscellaneous/structures
 
+    - repo: https://github.com/astral-sh/ruff-pre-commit
+      rev: v0.2.2
+      hooks:
+          - id: ruff-format
+            exclude: ^docs/.*
+          - id: ruff
+            args: [--fix, --exit-non-zero-on-fix, --show-fixes]
+
     - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
       rev: 0.2.3
       hooks:
           - id: yamlfmt
 
-    - repo: https://github.com/psf/black
-      rev: 23.12.1
-      hooks:
-          - id: black
-            language_version: python3 # Should be a command that runs python3.6+
-
-    - repo: https://github.com/PyCQA/flake8
-      rev: 6.1.0
-      hooks:
-          - id: flake8
-            args: [--count, --show-source, --statistics]
-            additional_dependencies:
-                - flake8-bugbear==23.2.13
-                - flake8-builtins==2.1.0
-                - flake8-comprehensions==3.10.1
-                - flake8-debugger==4.1.2
-                - flake8-logging-format==0.9.0
```

Comment:
I did not activate these two in ruff. We do not use logging in AWB and the flake8-debugger seems rather niche.